app/controllers/api/providers_controller.rb

```diff
@@ -106,8 +106,49 @@ def resume_resource(type, id, _data)
       action_result(false, "Could not resume Provider - #{err}")
     end
 
+    # Process change_password action for a single resource or a collection of resources
+    def change_password_resource(type, id, data = {})
+      if single_resource?
+        change_password(type, id, data)
+      else
+        change_password_multiple_providers(type, id, data)
+      end
+    end
+
     private
 
+    # Process password change request for a single resource
+    #
+    # @raise [BadRequestError] if no id is passed or some required data is missing
+    #                          or some error occur when try to change password on provider client
+    # @raise [NotFoundError]   if there isn't providers with the id passed
+    def change_password(type, id, data)
+      raise BadRequestError, "Must specify an id for change password of a #{type} resource" unless id
```

Comment:
I always have a hard time remembering how this path is reachable (IIRC we do some checking further up the stack) - can you add a test case for it, proving that it responded with this message?

Reply:
If you do a bulk request and not fill the `href` field. I don't think that it is verified previously, once I remove this verification no exception is thrown to the user. Anyway, I wrote a spec for it.